compiler/src/dmd/visitor.d

```diff
@@ -25,72 +25,74 @@ import dmd.root.rootobject;
+    void visit(AST.TypeInfoStaticArrayDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoAssociativeArrayDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoEnumDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoFunctionDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoDelegateDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoTupleDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoConstDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoInvariantDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoSharedDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoWildDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.TypeInfoVectorDeclaration s) { visit(cast(AST.TypeInfoDeclaration)s); }
+    void visit(AST.FuncAliasDeclaration s) { visit(cast(AST.FuncDeclaration)s); }
+    void visit(AST.ErrorInitializer i) { visit(cast(AST.Initializer)i); }
+    void visit(AST.ErrorExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.ComplexExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.StructLiteralExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.CompoundLiteralExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.ObjcClassReferenceExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.SymOffExp e) { visit(cast(AST.SymbolExp)e); }
+    void visit(AST.OverExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.HaltExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.DotTemplateExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.DotVarExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.DelegateExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.DotTypeExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.VectorExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.VectorArrayExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.SliceExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.ArrayLengthExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.DelegatePtrExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.DelegateFuncptrExp e) { visit(cast(AST.UnaExp)e); }
+    void visit(AST.DotExp e) { visit(cast(AST.BinExp)e); }
+    void visit(AST.IndexExp e) { visit(cast(AST.BinExp)e); }
+    void visit(AST.ConstructExp e) { visit(cast(AST.AssignExp)e); }
+    void visit(AST.BlitExp e) { visit(cast(AST.AssignExp)e); }
+    void visit(AST.RemoveExp e) { visit(cast(AST.BinExp)e); }
+    void visit(AST.ClassReferenceExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.VoidInitExp e) { visit(cast(AST.Expression)e); }
+    void visit(AST.ThrownExceptionExp e) { visit(cast(AST.Expression)e); }
 }
 
+alias Visitor = VisitorTemplate!ASTCodegen;
```

Comment:
I'm not convinced an alias is an appropriate solution for existing downstream users.

Reply:
I am not sure if there is a way to do that without changing existing code. I thought about a default template parameter value, a declaration like Visitor(AST = dmd.astcodegen.ASTCodegen) : ParseTimeVisitor!AST, but if we have only 1 template parameter it can't have a default value.

Reply:
If the problem is to have a true named symbol then instead of an alias:

```d
extern (C++) class Visitor : VisitorTemplate!ASTCodegen
{
    alias visit = typeof(this).visit;
}
```